### ISSUE TITLE
FileNotFoundException logged and reports are not sent

### ISSUE THREAD

**Issue by sfseyhan** [Instabug/Instabug-Android#137]
We get following warning in the logs when we try to build Instabug. Later on, when we try to send an email report, we get another warning in the logs even tho we get the success popup. No email is sent either.

### Stacktrace at building instabug
java.io.FileNotFoundException: /jacoco.exec (Read-only file system)
     at java.io.FileOutputStream.open0(Native Method)
     at java.io.FileOutputStream.open(FileOutputStream.java:308)
     at java.io.FileOutputStream.<init>(FileOutputStream.java:238)
     at org.jacoco.agent.rt.internal_8ff85ea.output.FileOutput.openFile(FileOutput.java:67)
     at org.jacoco.agent.rt.internal_8ff85ea.output.FileOutput.startup(FileOutput.java:49)
     at org.jacoco.agent.rt.internal_8ff85ea.Agent.startup(Agent.java:122)
     at org.jacoco.agent.rt.internal_8ff85ea.Agent.getInstance(Agent.java:50)
     at org.jacoco.agent.rt.internal_8ff85ea.Offline.<clinit>(Offline.java:31)
     at org.jacoco.agent.rt.internal_8ff85ea.Offline.getProbes(Offline.java:51)
     at com.instabug.library.Instabug$Builder.$jacocoInit(Unknown Source:13)
     at com.instabug.library.Instabug$Builder.<init>(Unknown Source:0)
     at com.takeaway.driver.DriverApplication.onCreate(DriverApplication.java:130)
     at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1154)
     at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5871)
     at android.app.ActivityThread.access$1100(ActivityThread.java:199)
     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1650)
     at android.os.Handler.dispatchMessage(Handler.java:106)
     at android.os.Looper.loop(Looper.java:193)
     at android.app.ActivityThread.main(ActivityThread.java:6669)
     at java.lang.reflect.Method.invoke(Native Method)
     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

### Stacktrace at sending report
java.io.FileNotFoundException: /storage/emulated/0/Android/data/com.takeaway.driver/files/instabug/attachments/bug_1540296646658_.jpg (No such file or directory)
     at java.io.FileInputStream.open0(Native Method)
     at java.io.FileInputStream.open(FileInputStream.java:231)
     at java.io.FileInputStream.<init>(FileInputStream.java:165)
     at com.instabug.library.util.BitmapUtils$1.run(BitmapUtils.java:111)
     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
     at com.instabug.library.util.threading.b$1.run(PriorityThreadFactory.java:28)
     at java.lang.Thread.run(Thread.java:764)

### SDK Version
8.0.11

### Android Version
Android 9

### Device Model
Google Pixel XL

**Comment by sfseyhan** [comment]
The reason why we are building instabug from activity is because we use a feature toggle to enable it at runtime.

If we build it at application onCreate with "InstabugInvocationEvent.NONE" and set another invocation mode later on, while building Instabug it doesn't throw the first warning in the logs but while sending reports the second FileNotFoundException is logged and reports are still not going through

**Comment by sfseyhan** [comment]
Correction: the first FileNotFoundException is logged as well at Instabug.Builder.build even if it's at Application.onCreate


W/System.err: java.io.FileNotFoundException: /jacoco.exec (Read-only file system)
2018-10-23 14:02:55.541 27588-27588/com.takeaway.driver W/System.err:     at java.io.FileOutputStream.open0(Native Method)
2018-10-23 14:02:55.542 27588-27588/com.takeaway.driver W/System.err:     at java.io.FileOutputStream.open(FileOutputStream.java:308)
2018-10-23 14:02:55.542 27588-27588/com.takeaway.driver W/System.err:     at java.io.FileOutputStream.<init>(FileOutputStream.java:238)
2018-10-23 14:02:55.542 27588-27588/com.takeaway.driver W/System.err:     at org.jacoco.agent.rt.internal_8ff85ea.output.FileOutput.openFile(FileOutput.java:67)
2018-10-23 14:02:55.542 27588-27588/com.takeaway.driver W/System.err:     at org.jacoco.agent.rt.internal_8ff85ea.output.FileOutput.startup(FileOutput.java:49)
2018-10-23 14:02:55.542 27588-27588/com.takeaway.driver W/System.err:     at org.jacoco.agent.rt.internal_8ff85ea.Agent.startup(Agent.java:122)
2018-10-23 14:02:55.542 27588-27588/com.takeaway.driver W/System.err:     at org.jacoco.agent.rt.internal_8ff85ea.Agent.getInstance(Agent.java:50)
2018-10-23 14:02:55.542 27588-27588/com.takeaway.driver W/System.err:     at org.jacoco.agent.rt.internal_8ff85ea.Offline.<clinit>(Offline.java:31)
2018-10-23 14:02:55.542 27588-27588/com.takeaway.driver W/System.err:     at org.jacoco.agent.rt.internal_8ff85ea.Offline.getProbes(Offline.java:51)
2018-10-23 14:02:55.543 27588-27588/com.takeaway.driver W/System.err:     at com.instabug.library.Instabug$Builder.$jacocoInit(Unknown Source:13)
2018-10-23 14:02:55.543 27588-27588/com.takeaway.driver W/System.err:     at com.instabug.library.Instabug$Builder.<init>(Unknown Source:0)
2018-10-23 14:02:55.543 27588-27588/com.takeaway.driver W/System.err:     at com.takeaway.driver.DriverApplication.onCreate(DriverApplication.java:130)
2018-10-23 14:02:55.543 27588-27588/com.takeaway.driver W/System.err:     at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1154)
2018-10-23 14:02:55.543 27588-27588/com.takeaway.driver W/System.err:     at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5871)
2018-10-23 14:02:55.543 27588-27588/com.takeaway.driver W/System.err:     at android.app.ActivityThread.access$1100(ActivityThread.java:199)
2018-10-23 14:02:55.543 27588-27588/com.takeaway.driver W/System.err:     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1650)
2018-10-23 14:02:55.543 27588-27588/com.takeaway.driver W/System.err:     at android.os.Handler.dispatchMessage(Handler.java:106)
2018-10-23 14:02:55.543 27588-27588/com.takeaway.driver W/System.err:     at android.os.Looper.loop(Looper.java:193)
2018-10-23 14:02:55.544 27588-27588/com.takeaway.driver W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:6669)
2018-10-23 14:02:55.544 27588-27588/com.takeaway.driver W/System.err:     at java.lang.reflect.Method.invoke(Native Method)
2018-10-23 14:02:55.544 27588-27588/com.takeaway.driver W/System.err:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
2018-10-23 14:02:55.544 27588-27588/com.takeaway.driver W/System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

**Comment by sfseyhan** [comment]
We have noticed our trial period has ended and wondering if this can be the reason why report emails are no longer going through. If that's the case, there should be proper error handling, or give the app responsibility to handle such error, instead of success message, giving the impression to user that he has sent the report.

**Comment by sfseyhan** [comment]
After our trial period got extended, report emails started to be delivered. The warnings are still logged so I guess they were irrelevant. We would still appreciate an error handling in that situation.

**Comment by rdiaa** [comment]
Hello, @sfseyhan I'm so so sorry that you are facing this but I appreciate if you can tell me know what is the problem you are facing right now?

**Comment by sfseyhan** [comment]
Right now I am not facing a problem. I didn't like trying to figure out why report sending is not working because there was no indication in the logs about expiration of our trial period.

What I am complaining about is we were getting success message and no hint of what is actually wrong with reporting issues, causing us hours of tinkering with how we use the sdk and trying different versions.

### Steps to Reproduce the Problem
- Use an account with expired trial period.
- Report an issue with instabug
- Success message is received suggesting the issue report is sent.
- No report was sent.

### Expected Behavior
- No success message should be shown if the operation is not successful.
- Proper error handling identifying what is wrong. Display an error message, or have an error callback for client app to deal with it.

### Actual Behavior
- No error message, no error logs, no error callback. Causing hours of investigation what might be wrong.

**Comment by rdiaa** [comment]
Hello, @sfseyhan, first of all, thank you so much for the detailed information provided. I will check with the Products team and get back to you. Thank you again for spotting this.

**Comment by rdiaa** [comment]
Hello again, 

I just want to inform you that when your trial ends you are automatically directed to free plan and by all means, you should receive your bugs If you are not receiving them, then we should check the problem. Also, I want to tell you that when you are in the free plan you are eligible for one-month data retention. So after this, you will find that some bugs are missing and that's because of data retention. Please let me know if you are not receiving the bugs so I can check it now.

**Comment by sfseyhan** [comment]
@rdiaa We were not receiving the issues reported from android while our trial period was expired, but ios was still able to send them. Once I requested the trial to be extended a week, reports started to go through from android as well. We don't have an issue at the moment but if the trial ends it might occur again.

Not sure if it's related but when trial period ended, dashboard wasn't accessible because we added 4 team members which the free plan doesn't support.

**Comment by NeverMorer** [comment]
same issue on 8.0.11

**Comment by NeverMorer** [comment]
@sfseyhan i got the error cause i  build twice carelessly

**Comment by rdiaa** [comment]
@NeverMorer @dirkam @sfseyhan  What error are you facing?

**Comment by dirkam** [comment]
@rdiaa 
```java
java.io.FileNotFoundException: /jacoco.exec (Read-only file system)
             at java.io.FileOutputStream.open0(Native Method)
             at java.io.FileOutputStream.open(FileOutputStream.java:308)
             at java.io.FileOutputStream.<init>(FileOutputStream.java:238)
             at org.jacoco.agent.rt.internal_8ff85ea.output.FileOutput.openFile(FileOutput.java:67)
             at org.jacoco.agent.rt.internal_8ff85ea.output.FileOutput.startup(FileOutput.java:49)
             at org.jacoco.agent.rt.internal_8ff85ea.Agent.startup(Agent.java:122)
             at org.jacoco.agent.rt.internal_8ff85ea.Agent.getInstance(Agent.java:50)
             at org.jacoco.agent.rt.internal_8ff85ea.Offline.<clinit>(Offline.java:31)
             at org.jacoco.agent.rt.internal_8ff85ea.Offline.getProbes(Offline.java:51)
             at com.instabug.library.Instabug$Builder.$jacocoInit(Unknown Source:13)
             at com.instabug.library.Instabug$Builder.<init>(Unknown Source:0)
             at MyApplicationClass.onCreate(MyApplicationClass.java:61)
             at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1154)
             at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5871)
             at android.app.ActivityThread.access$1100(ActivityThread.java:199)
             at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1650)
             at android.os.Handler.dispatchMessage(Handler.java:106)
             at android.os.Looper.loop(Looper.java:193)
             at android.app.ActivityThread.main(ActivityThread.java:6669)
             at java.lang.reflect.Method.invoke(Native Method)
             at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by dirkam** [comment]
I also see the other stracktrace before bug reports are sent. Though the bug reports show up in the dashboard.

```java
java.io.FileNotFoundException: /storage/emulated/0/Android/data/my.package/files/instabug/attachments/bug_123456789_.jpg (No such file or directory)
             at java.io.FileInputStream.open0(Native Method)
             at java.io.FileInputStream.open(FileInputStream.java:231)
             at java.io.FileInputStream.<init>(FileInputStream.java:165)
             at com.instabug.library.util.BitmapUtils$1.run(BitmapUtils.java:111)
             at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
             at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
             at com.instabug.library.util.threading.b$1.run(PriorityThreadFactory.java:28)
             at java.lang.Thread.run(Thread.java:764)
```

**Comment by rdiaa** [comment]
@dirkam we are working on the fix and will update you once its released

**Comment by rdiaa** [comment]
@dirkam I just want to inform you that the fix should be released within two weeks with the new version. I will update you once it's released

**Comment by matir91** [comment]
@rdiaa Do you know which is the latest working version of the sdk? Because right now it's not working and I cannot wait 2 weeks.

**Comment by rdiaa** [comment]
@matir91  you mean related to the same issue? The latest version is 8.0.11 and we  will release the fix of this issue next week

**Comment by rdiaa** [comment]
@matir91 @dirkam  I want to inform you that the latest version now is released. It's version 8.12. Please update the SDK, Everything should be working fine. Please feel free to re-open the thread if anything is not in place.

**Comment by amitav13** [comment]
I'm still seeing this with 8.0.14.

**Comment by rdiaa** [comment]
@amitav13  can you please check this snap-shot till this fix is released.

**Comment by amitav13** [comment]
@rdiaa what's the repo URL for accessing the snapshot? Studio couldn't resolve it from:
```
google()        
jcenter()
mavenCentral()
maven { url "https://jitpack.io" }
```

**Comment by rdiaa** [comment]
@amitav13 
To use the snapshot:
Add these to your app-level build.gradle file to grab snapshots:


dependencies {
 compile 'com.instabug.library:instabug:`8.0.11.13-SNAPSHOT`'
}

repositories {
   
maven {
       
url "https://oss.sonatype.org/content/repositories/snapshots"
   
}
}

**Comment by amitav13** [comment]
@rdiaa the snapshot version seems to fix it!

**Comment by rdiaa** [comment]
Glad to hear that

**Comment by ismaeldosil** [comment]
@rdiaa this fix is in any stable version?

**Comment by mzelzoghbi** [comment]
Hi @ismaeldosil, This was included in latest release **8.0.19**

```
implementation 'com.instabug.library:instabug:8.0.19'
```

Please feel free to reopen this thread again if you need further help with anything else. Have a good day!

**Comment by ismaeldosil** [comment]
https://github.com/Instabug/Instabug-Android/issues/137#issuecomment-462861127


> Hi @ismaeldosil, This was included in latest release **8.0.19**
> 
> ```
> implementation 'com.instabug.library:instabug:8.0.19'
> ```
> Please feel free to reopen this thread again if you need further help with anything else. Have a good day!

Hey @mzelzoghbi, Im using instabug 8.1.= and I still seeing ths issue.
Im using this configuration:

Could you reopen this?

```java
Instabug instabug = new Instabug.Builder(application, instabugApplicationId)
        .setInvocationEvents(enableInstabugFlatingButton ? InstabugInvocationEvent.FLOATING_BUTTON : InstabugInvocationEvent.NONE)
        .setInAppMessagingState(Feature.State.DISABLED)
        .setTrackingUserStepsState(Feature.State.ENABLED)
        .setConsoleLogState(Feature.State.ENABLED)
        .setInstabugLogState(Feature.State.ENABLED)
        .setReproStepsState(State.ENABLED)
        .setViewHierarchyState(Feature.State.ENABLED)
        .setUserDataState(Feature.State.ENABLED)
        .setUserEventsState(Feature.State.ENABLED)
        .build();

      //Instabug 8.1 API changes
      Surveys.setState(Feature.State.ENABLED);
      Replies.setPushNotificationState(Feature.State.ENABLED);
      CrashReporting.setState(Feature.State.ENABLED);

      // Disable Welcome dialog
      Instabug.showWelcomeMessage(WelcomeMessage.State.DISABLED);
      Instabug.setWelcomeMessageState(WelcomeMessage.State.DISABLED);

      return instabug;
```
